test/e2e/upgrade_ironic_test.go

```diff
@@ -0,0 +1,59 @@
+package e2e
+
+import (
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+
+	appsv1 "k8s.io/api/apps/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+func upgrade_ironic() {
+	clientSet := targetCluster.GetClientSet()
+	ironicDeployName := "baremetal-operator-ironic"
+	By("Upgrading ironic image based containers")
+	getIronicDeployment := func() (*appsv1.Deployment, error) {
+		return clientSet.AppsV1().Deployments("baremetal-operator-system").Get(ctx, ironicDeployName, metav1.GetOptions{})
+	}
```

Comment:
I think it would be a bit more natural if `getIronicDeployment` were defined with a top level `func` statement rather than assigning an anonymous function. What do you think?